File: Array/Array-101/Problems/difference_pairs/difference_pairs.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int x, n, diff;

    // Ask the user for the size of the array
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    // Declare an array of size n
    int arr[n];

    // Input elements into the array
    printf("Enter the elements of the array: ");
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }

    // Display the original array
    printf("The elements of the array are: ");
    for (int i = 0; i < n; i++)
    {
        printf("%d ", arr[i]);
    }

    // Ask the user to enter a positive number for difference
    printf("\nEnter a positive number: ");
    scanf("%d", &x);

    // Print pairs of elements with the given difference
    printf("The following pairs of elements give us a difference of %d:\n", x);
    for (int i = 0; i < n - 1; i++)
    {
        for (int j = i + 1; j < n; j++)
        {
            // Calculate absolute difference between arr[i] and arr[j]
            diff = abs(arr[j] - arr[i]);

            // Check if the absolute difference matches x
            if (diff == x)
            {
                printf("(%d, %d);", arr[i], arr[j]);
            }
        }
    }
}

File: Array/Array-101/Problems/difference_pairs/difference_pairs.py
def main():
    # Input the size of the array
    n = int(input("Enter the size of the list: "))

    # Input the elements
    arr = []
    print("Enter the elements of the list:")
    for _ in range(n):
        arr.append(int(input()))

    # Display the original list
    print("The elements of the list are:", arr)

    # Input the target difference
    x = int(input("Enter a positive number: "))

    # Find and print pairs with the given difference
    print(f"The following pairs of elements give us a difference of {x}:")
    for i in range(n - 1):
        for j in range(i + 1, n):
            if abs(arr[j] - arr[i]) == x:
                print(f"({arr[i]}, {arr[j]});", end=' ')

if __name__ == "__main__":
    main()
